### DISCUSSION TITLE
Translation Strings Get Overwritten When Two Users Edit Concurrently

### DISCUSSION THREAD

**Discussion by obezzad** [directus/directus#14056] [Archive]
Given that at least two users are working on the translation strings simultaneously, when a user (A) does any change, it overwrites all the changes done by others between the last refresh of `/admin/settings/translation-strings` by the user (A) and the time when the user (A) made any change (we observed this issue when adding new translation strings concurrently).

This issue seems to occur regardless of the environment (node version, DBMS, OS, browser...).

### To Reproduce

1. Open the `/admin/settings/translation-strings` page using two users A and B in two different sessions (browsers, incognito...)
2. Add a translation string using user A
3. Without refreshing the page in the other session, add a translation string using user B
4. Refresh the page in user A, and you'll notice that the translation string done by user A was overwritten by B's change

**Comment by azrikahar** [comment]
I believe this is essentially the aim of #4793, as translation strings are technically a single field on the DB side. Similar to the question raised in #13839, specifically the answer https://github.com/directus/directus/discussions/13839#discussioncomment-2925375:

> If both users edit `title` simultaneously, whichever edit is saved last is the one that's saved to the database

The Websockets feature on the roadmap should help pave the way to alleviate such scenarios.

**Reply by obezzad** [reply]
Makes sense. Thank you!
